s3client.go

```diff
@@ -184,15 +185,21 @@ func (client *s3client) UploadFile(bucketName string, remotePath string, localPa
 	}
 
 	defer localFile.Close()
-	
+
 	// Automatically adjust partsize for larger files.
 	fSize := stat.Size()
-	if fSize > int64(uploader.MaxUploadParts) * uploader.PartSize {
-		partSize := fSize / int64(uploader.MaxUploadParts)
-		if fSize % int64(uploader.MaxUploadParts) != 0 {
-			partSize++
+	if !options.DisableMultipart {
+		if fSize > int64(uploader.MaxUploadParts)*uploader.PartSize {
+			partSize := fSize / int64(uploader.MaxUploadParts)
+			if fSize%int64(uploader.MaxUploadParts) != 0 {
+				partSize++
+			}
+			uploader.PartSize = partSize
 		}
-		uploader.PartSize = partSize
+	} else {
+		uploader.MaxUploadParts = 1
+		uploader.Concurrency = 1
+		uploader.PartSize = fSize + 1
```

Comment:
Why `fSize + 1` here and not just `fSize`?

Was setting `uploader.MaxUploadParts` to 1 not sufficient? Setting concurrency and part size seems like it may be redundant.

Reply:
I do not know in the latest version, but in this aws sdk version of resouce, setting `Concurrency` to 1 does not seem to adjust `MaxUploadParts` accordingly.

An error occurred when only one of `MaxUploadParts` and `Concurrency` was used in my environment.

Even though they duplicate, I thought that it would be better not to occur an error.

And, an error occurred when just `fSize`....

```
5.81 MB / 5.81 MB [====================================] 100.00 % 149.08 MB/s 0serror running command: MethodNotAllowed: The specified method is not allowed against this resource.
	status code: 405, request id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx, host id: 
```

Considering these, as a result of trial and error, it became like the present code.